cmake/compiler_settings.cmake

```diff
@@ -70,6 +70,13 @@ else ()
 endif ()
 
 if (CMAKE_CXX_COMPILER_ID MATCHES GNU OR CMAKE_CXX_COMPILER_ID MATCHES Clang)
+  # Enforce -O0 in debug mode.
+  string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_UPPER)
+  if (CMAKE_BUILD_TYPE_UPPER STREQUAL "DEBUG")
+    set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0")
```

Comment:
Should we also add /Od for Windows to disable optimization for Windows host code?

Reply:
Yes this option is already there. https://github.com/openenclave/openenclave/blob/d2050f5c2ac0caa7cf03eaf1b1adc3ddc9d6a5a3/cmake/compiler_settings.cmake#L100